Detectors/EMCAL/workflow/src/RecoWorkflow.cxx

```diff
@@ -0,0 +1,155 @@
+                                                     "o2sim",
+                                                     { "digitbranch", "EMCALDigit", "Digit branch" },
+                                                     { "mcbranch", "EMCALDigitMCTruth", "MC label branch" },
+                                                     o2::framework::OutputSpec{ "EMC", "DIGITS" },
+                                                     o2::framework::OutputSpec{ "EMC", "DIGITSMCTR" } },
+                                                   propagateMC));
+
+    specs.emplace_back(o2::emcal::reco_workflow::getEmcalDigitsPrinterSpec());
+  }
+
+  // check if the process is ready to quit
+  // this is decided upon the meta information in the EMCAL block header, the operation is set
+  // value kNoPayload in case of no data or no operation
+  // see also PublisherSpec.cxx
+  // in this workflow, the EOD is sent after the last real data, and all inputs will receive EOD,
+  // so it is enough to check on the first occurence
+  // FIXME: this will be changed once DPL can propagate control events like EOD
+  auto checkReady = [](o2::framework::DataRef const& ref, bool& isReady) {
+    LOG(DEBUG) << "[checkReady] called";
+    auto const* emcalheader = o2::framework::DataRefUtils::getHeader<o2::emcal::EMCALBlockHeader*>(ref);
+    // sector number -1 indicates end-of-data
+    if (emcalheader != nullptr) {
+      if (emcalheader->mPayloadType == o2::emcal::EMCALBlockHeader::PayloadType_t::kDigits) {
+        LOG(DEBUG) << "[checkReady] Payload is digits";
+      } else if (emcalheader->mPayloadType == o2::emcal::EMCALBlockHeader::PayloadType_t::kNoPayload) {
+        LOG(DEBUG) << "[checkReady] Payload is undefined";
+      }
+      isReady = emcalheader->mPayloadType == o2::emcal::EMCALBlockHeader::PayloadType_t::kNoPayload;
+      // indicate normal processing if not ready and skip if ready
+      if (isReady) {
+        LOG(DEBUG) << "[checkReady] No longer ready";
+        return o2::framework::MakeRootTreeWriterSpec::TerminationCondition::Action::SkipProcessing;
+      }
+    } else {
+      LOG(DEBUG) << "[checkReady] No header received";
+    }
+    LOG(DEBUG) << "[checkReady] Ready - do normal processing ...";
+    return o2::framework::MakeRootTreeWriterSpec::TerminationCondition::Action::DoProcessing;
+  };
+
+  auto makeWriterSpec = [propagateMC, checkReady](const char* processName, const char* defaultFileName, const char* defaultTreeName,
+                                                  auto&& databranch, auto&& mcbranch) {
+    // depending on the MC propagation flag, the RootTreeWriter spec is created with two
+    // or one branch definition
+    if (propagateMC) {
+      return std::move(o2::framework::MakeRootTreeWriterSpec(processName, defaultFileName, defaultTreeName,
+                                                             o2::framework::MakeRootTreeWriterSpec::TerminationCondition{ checkReady },
```

Comment:
```suggestion
                                                             1, // default number of events
```

Reply:
Shouldn't the publisher be able to publish more than just 1 event? I would prefer being able to send a variable number of events, ideally as much as in the tree.

Reply:
you always have the possibility to set the number of events with the `nevents` command line option